libqtile/widget/base.py

```diff
@@ -388,13 +388,43 @@ class _TextBox(_Widget):
         ("markup", True, "Whether or not to use pango markup"),
         ("fmt", "{}", "How to format the text"),
         ("max_chars", 0, "Maximum number of characters to display in widget."),
+        ("enable_scrolling", False, "Whether text should be scrolled"),
+        (
+            "repeat_scrolling",
+            False,
+            "Whether text should restart scrolling once the text has ended",
+        ),
+        (
+            "start_end_delay",
+            2,
+            "Number of seconds to pause before starting scrolling and restarting/clearing text at end",
+        ),
+        ("scroll_step", 1, "Number of pixels to scroll with each step"),
+        ("scroll_interval", 0.1, "Time in seconds before next scrolling step"),
+        ("scroll_width", 200, "Width of widget. Text wider than this will scroll."),
+        (
+            "scroll_clear",
+            False,
+            "Whether text should scroll completely away (True) or stop when the end of the text is shown (False)",
+        ),
+        ("reset_scroll_on_update", True, "When new text is provided, the scroll is restarted"),
+        ("hide_text_after_scroll", False, "Whether the widget should hide when scrolling has finised")
```

Comment:
It would help with clarity if all scroll-related settings had the same prefix, like the ones with `scroll_X`. Maybe `enabled_scrolling` can just be `scroll`, then `scroll_repeat`, `scroll_reset_on_update`, `scroll_hide_when_done` or something.

re the settings themselves, are these all settings that you think would be immediately useful (like in test widgets you may have configured), or are you predicting what might be useful? It's just I'm a bit skeptical about the usefulness of e.g. `repeat_scrolling` (wouldn't everybody want `True`?), `reset_scroll_on_update` (same). The role of `scroll_clear` and `hide_text_after_scroll` also seems similar - scroll once, then disappear, just that ones scrolls away rather than vanishes? How do these interact with `repeat_scrolling`, and do the constraints on how they can be used together suggest that the configuration options can be made more concise?

Lastly, is `scroll_width` not redundant with `width`, as only one can be used, but conceptually they mean something similar? Could `width` be used the way `scroll_width` is being used when `enable_scrolling = True`?

Reply:
Thanks. All good points. 

This is what I started with to get things working but I agree we can probably tidy up a bit.

I'm just playing with mpris2 widget but can refactor after that.

Reply:
`scroll_clear`: when `True` this keeps scrolling until all the text has scrolled out of the widget. When `False` the scrolling stops as soon as the end of the text is visible.
`hide_text_after_scroll`: This scrolls once and then hides the text.

Reply:
Made some changes. Haven't changed `width` yet. One reason for this is that it doesn't appear in `defaults` and I wanted to make it clearer but I definitely see the value in your approach too.